ecs-init/config/common.go

```diff
@@ -212,6 +218,23 @@ func DockerUnixSocket() (string, bool) {
 	return "/var/run", false
 }
 
+// CredentialsFetcherUnixSocket returns the credentials fetcher daemon socket endpoint and whether it reads from CredentialsFetcherEnvVar
+func CredentialsFetcherUnixSocket() (string, bool) {
+	if credentialsFetcherHost := os.Getenv(CredentialsFetcherHostEnvVar); strings.HasPrefix(credentialsFetcherHost, UnixSocketPrefix) {
+		return strings.TrimPrefix(credentialsFetcherHost, UnixSocketPrefix), true
+	}
+
+	return "/var/credentials-fetcher/socket/credentials_fetcher.sock", false
+}
+
+// HostCredentialsFetcherPath() returns the daemon socket lacation
+func HostCredentialsFetcherPath() string {
```

Comment:
should this also return (string, bool) rather than "" as the missing indicator?

Reply:
following the pattern of other Host methods

Reply:
I also agree with Ray. returning string, bool seems much safer than checking for empty strings.

Reply:
updated accordingly